### ISSUE TITLE
Rack::Static - URL prefixes ignored when index used

### ISSUE THREAD

**Issue by erwanlr** [rack/rack#1452]
Using Rack::Static, when the index option is used, the URL prefixes seem to be ignored, which leads to content being served when it should not.

Example:

```
 ├── static/
 │   ├── css/
 |   |   ├── style.css
 |   |   └── index.html
 │   ├── feed/
 |   |   └── index.html
 |   └── index.html
 └── config.ru
```

```ruby
# config.ru
# frozen_string_literal: true

use Rack::Static, urls: %w[/css], root: 'static', index: 'index.html'

run lambda { |_| [404, {}, []] }
```

## Expected behaviour
Request | Response
---------|----------
GET / | 404 Not Found
GET /feed/ | 404 Not Found
GET /css/ | Content of static/css/index.html
GET /css/style.css | Content of static/css/style.css

## Actual behaviour
Request | Response
---------|----------
GET / | Content of static/index.html
GET /feed/ | Content of static/feed/index.html
GET /css/ | Content of static/css/index.html
GET /css/style.css | Content of static/css/style.css

(Seems to be related to https://github.com/rack/rack/issues/740)

**Comment by ioquatix** [comment]
Can you make a PR?

**Comment by jeremyevans** [comment]
I tried to reproduce this issue using the provided code and directory structure and was unable to do so:

```
$ find static
static
static/css
static/css/style.css
static/css/index.html
static/feed
static/feed/index.html
static/index.html
$ curl -i http://127.0.0.1:9292/              
HTTP/1.1 404 Not Found
Transfer-Encoding: chunked

$ curl -i http://127.0.0.1:9292/feed/         
HTTP/1.1 404 Not Found
Transfer-Encoding: chunked

$ curl -i http://127.0.0.1:9292/css/          
HTTP/1.1 200 OK
Last-Modified: Sat, 11 Jan 2020 02:39:42 GMT
Content-Type: text/html
Content-Length: 10

css-index
$ curl -i http://127.0.0.1:9292/css/style.css 
HTTP/1.1 200 OK
Last-Modified: Sat, 11 Jan 2020 02:39:19 GMT
Content-Type: text/css
Content-Length: 4

css

```

The above output was produced using Puma as the web server.  Webrick and Unicorn produced similar results.

If you still think this is a bug in rack, please submit a failing test.

**Comment by erwanlr** [comment]
The issue appears to have been fixed in Rack 2.1.0 released earlier.

Below was my output with Rack 2.0.8

```
$ find static
static
static/index.html
static/css
static/css/index.html
static/css/style.css
static/feed
static/feed/index.html
```

```
$ rackup --version
Rack 1.3 (Release: 2.0.8)
```

```
$ cat config.ru

# config.ru
# frozen_string_literal: true

use Rack::Static, urls: %w[/css], root: 'static', index: 'index.html'

run lambda { |_| [404, {}, []] }
```

```
$ curl -i http://127.0.0.1:9292/
HTTP/1.1 200 OK 
Last-Modified: Tue, 07 Jan 2020 12:45:37 GMT
Content-Type: text/html
Content-Length: 12
Server: WEBrick/1.4.2 (Ruby/2.6.5/2019-10-01)
Date: Sat, 11 Jan 2020 07:16:21 GMT
Connection: Keep-Alive

/index.html
```

```
$ curl -i http://127.0.0.1:9292/feed/
HTTP/1.1 200 OK 
Last-Modified: Tue, 07 Jan 2020 12:46:35 GMT
Content-Type: text/html
Content-Length: 17
Server: WEBrick/1.4.2 (Ruby/2.6.5/2019-10-01)
Date: Sat, 11 Jan 2020 07:17:12 GMT
Connection: Keep-Alive

/feed/index.html
```

```
$ curl -i http://127.0.0.1:9292/css/
HTTP/1.1 200 OK 
Last-Modified: Tue, 07 Jan 2020 12:46:27 GMT
Content-Type: text/html
Content-Length: 16
Server: WEBrick/1.4.2 (Ruby/2.6.5/2019-10-01)
Date: Sat, 11 Jan 2020 07:17:54 GMT
Connection: Keep-Alive

/css/index.html
```

```
$ curl -i http://127.0.0.1:9292/css/style.css
HTTP/1.1 200 OK 
Last-Modified: Tue, 07 Jan 2020 12:46:19 GMT
Content-Type: text/css
Content-Length: 15
Server: WEBrick/1.4.2 (Ruby/2.6.5/2019-10-01)
Date: Sat, 11 Jan 2020 07:18:06 GMT
Connection: Keep-Alive

/css/style.css
```

```
$ rackup 
[2020-01-11 08:15:57] INFO  WEBrick 1.4.2
[2020-01-11 08:15:57] INFO  ruby 2.6.5 (2019-10-01) [x86_64-darwin18]
[2020-01-11 08:15:57] INFO  WEBrick::HTTPServer#start: pid=92516 port=9292
127.0.0.1 - - [11/Jan/2020:08:16:21 +0100] "GET /index.html HTTP/1.1" 200 12 0.0013
127.0.0.1 - - [11/Jan/2020:08:17:12 +0100] "GET /feed/index.html HTTP/1.1" 200 17 0.0033
127.0.0.1 - - [11/Jan/2020:08:17:54 +0100] "GET /css/index.html HTTP/1.1" 200 16 0.0016
127.0.0.1 - - [11/Jan/2020:08:18:06 +0100] "GET /css/style.css HTTP/1.1" 200 15 0.0019
```

**Comment by ioquatix** [comment]
Okay great, if you are satisfied the issue is fixed can you please close the issue :)
